### PR TITLE
CLN: simplify MultiIndex._shallow_copy

### PR DESCRIPTION
**Pull request by topper-123**
Minor simplification of ``MultiIndex._shallow_copy``.

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -990,15 +990,11 @@ def _constructor(self):
     def _shallow_copy(self, values=None, **kwargs):
         if values is not None:
             names = kwargs.pop("names", kwargs.pop("name", self.names))
-            # discards freq
-            kwargs.pop("freq", None)
```
Comment (topper-123) [inline pandas/core/indexes/multi.py]: This makes no sense, ``freq`` isn't accepted by any MultiIndex constructor.

Comment (jbrockmendel) [reply pandas/core/indexes/multi.py]: +1

**Comment by jbrockmendel** [comment]
LGTM

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -990,15 +990,11 @@ def _constructor(self):
     def _shallow_copy(self, values=None, **kwargs):
         if values is not None:
             names = kwargs.pop("names", kwargs.pop("name", self.names))
```
Comment (jbrockmendel) [inline pandas/core/indexes/multi.py]: do we need to worry about both name and names being passed?

Comment (topper-123) [reply pandas/core/indexes/multi.py]: ``MultiIndex.from_tuples`` only accepts ``names``, and Mi uses ``name`` in some locations, so yes (but it a part of the ickyness of MultiIndex, so should ideally be refactored away).

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**